srunner/scenarios/opposite_vehicle_taking_priority.py

```diff
@@ -133,24 +133,44 @@ def _create_behavior(self):
             self.other_actors[0], self.ego_vehicle, location_of_collision_dynamic)
         sync_arrival_stop = InTriggerDistanceToVehicle(self.other_actors[0],
                                                        self.ego_vehicle,
-                                                       15)
+                                                       41)
         sync_arrival_parallel.add_child(sync_arrival)
         sync_arrival_parallel.add_child(sync_arrival_stop)
 
-        keep_velocity_for_distance = py_trees.composites.Parallel(
-            "Keep velocity for distance",
+        # Generate plan for WaypointFollower
+        waypoint = self.other_actors[0].get_world().get_map().get_waypoint(self.other_actors[0].get_location())
+
+        turn = 0  # drive straight ahead
+        plan = []
+
+        # generating waypoints until intersection (target_waypoint)
+        plan, target_waypoint = generate_target_waypoint_list(
+            CarlaDataProvider.get_map().get_waypoint(self.other_actors[0].get_location()), turn)
+
+        # Generating waypoint list till next intersection
+        wp_choice = target_waypoint.next(5.0)
+        while len(wp_choice) == 1:
+            target_waypoint = wp_choice[0]
+            plan.append((target_waypoint, RoadOption.LANEFOLLOW))
+            wp_choice = target_waypoint.next(5.0)
+
+        continue_driving = py_trees.composites.Parallel(
+            "ContinueDriving",
             policy=py_trees.common.ParallelPolicy.SUCCESS_ON_ONE)
-        keep_velocity = KeepVelocity(
-            self.other_actors[0],
-            self._other_actor_target_velocity)
-        keep_velocity_distance = DriveDistance(
+
+        continue_driving_waypoints = WaypointFollower(
+            self.other_actors[0], self._other_actor_target_velocity, plan=plan)
+
+        continue_driving_distance = DriveDistance(
             self.other_actors[0],
             self._other_actor_distance,
             name="Distance")
-        keep_velocity_timeout = TimeOut(5)
-        keep_velocity_for_distance.add_child(keep_velocity)
-        keep_velocity_for_distance.add_child(keep_velocity_distance)
-        keep_velocity_for_distance.add_child(keep_velocity_timeout)
+
+        continue_driving_timeout = TimeOut(10)
+
+        continue_driving.add_child(continue_driving_waypoints)
+        continue_driving.add_child(continue_driving_distance)
```

Comment:
Currently, if the ego vehicle decides to drive in the same direction as adversary, the ego vehicle is stuck behind the adversary vehicle, it has to change lane to complete the scenario. Is it intended? Otherwise scenario ends with a timeout as the end condition of the scenario doesn't meet.

Reply:
Yes correct. The only chance I see to avoid this, would be to feed the ego route into the scenario. This however, would require some re-design.

For now, I think we can still use this scenario. Even if the adversary takes the same route, it will be in front of the ego vehicle (so a follow maneuver), and disappear as soon as this scenario is over. The disappearance could be made even a bit more aggressively.

So, all in all, my suggestion would be to keep it as it is for now. If our testing reveals that it is a real problem, we can come back and work on it.